plugins/org.jboss.tools.openshift.egit.core/src/org/jboss/tools/openshift/egit/core/EGitUtils.java

```diff
@@ -229,7 +229,11 @@ public static Repository createRepository(IProject project,
 	}
 
 	public static File getRepositoryPathFor(IProject project) {
-		return new File(project.getLocationURI().getPath(), Constants.DOT_GIT);
+		File f = new File(project.getLocationURI().getPath());
```

Comment:
This is exactly what EGit RepositoryFinder#indInDirectoryAndParents already does in a more sophisticated manner:
https://github.com/eclipse/egit/blob/master/org.eclipse.egit.core/src/org/eclipse/egit/core/project/RepositoryFinder.java#L170